language/move-lang/tests/move_check/parser/spec_parsing_ok.move

```diff
@@ -0,0 +1,101 @@
+// These are some basic parsing tests for specifications which are expected to succeed.
+// Full testing of spec parsing correctness is done outside of this crate.
+//
+// Note that even though we only test parsing, we still need to ensure that the move code (not the specification)
+// is type checking correctly, because with no parsing error, the test harness
+// will run subsequent phases of the move-lang compiler.
+//
+// For parse failures, see the `spec_*_fail.move` test cases.
+
+module M {
+    synthetic expected_coin_sum: u64;
+    synthetic other: bool;
+
+    resource struct T{x: u64}
+    resource struct R{x: u64}
+
+    struct SomeCoin {
+        x : u64,
+        invariant x > 0,
+        y: u64,
+        invariant x == y,
+        invariant update old(y) < x,  // this weird coin only allows to increase it's value
+
+        // Tracking synthetic variable use case.
+        invariant update expected_coin_sum = expected_coin_sum - old(x) + x,
```

Comment:
I think we should put these invariants outside of the struct somehow. Either pre- or post-fix. 

Having them inside feels weird to me. The current syntax for Structs is intended to give this good feeling of symmetry between declaration/packing/unpacking. I feel that having the invariants inside breaks that sort of feeling. 
In other words, the invariants aren't "fields" in the struct, so they should be outside of the curly braces

Reply:
We now have the spec blocks (and outside of struct)